lib/bot.js

```diff
@@ -179,7 +188,23 @@ class Bot {
       }
 
       const withMentions = text.replace(/@[^\s]+\b/g, match => {
-        const user = this.discord.users.get('username', match.substring(1));
+        let user = this.discord.users.get('username', match.substring(1));
+        const server = discordChannel.server;
+        if (user) {
+          if (this.getDiscordNicknameOnServer(user, server).toUpperCase()
+          !== match.substring(1).toUpperCase()) {
+            user = null; // Nickname doesn't match mention
+          }
+        }
+        if (!user) {
+          const length = this.discord.users.length;
+          for (let i = 0; i < length && !user; i++) {
+            if (this.getDiscordNicknameOnServer(this.discord.users[i], server).toUpperCase()
```

Comment:
Wouldn't this create an API call for every discord user on the server, and do that for every single person someone has mentioned?